c10/util/TypeCast.h

```diff
@@ -68,6 +69,36 @@ struct static_cast_with_inter_type<uint8_t, src_t> {
   }
 };
 
+template <>
+struct static_cast_with_inter_type<c10::complex<c10::Half>, c10::BFloat16> {
+  C10_HOST_DEVICE __ubsan_ignore_undefined__ static inline c10::complex<
+      c10::Half>
+  apply(c10::BFloat16 src) {
+    return static_cast<c10::complex<c10::Half>>(c10::complex<float>{src});
```

Comment:
```suggestion
    return static_cast<c10::complex<c10:: BFloat16 >>(c10::complex<float>{src});
```

Reply:
It should be `c10::complex<c10::Half>`, right? Our complex32 will correspond to only that. So this will allow casting from `BFloat16` to `complex<Half>`.

Reply:
you are totally right I got confused